### DISCUSSION TITLE
getSession and session callback

### DISCUSSION THREAD

**Discussion by Mosquid** [nextauthjs/next-auth#2841] [Help]
I use a function like this in my `getServerSideProps`:

```
const sessionData = await getSession();
const token = await getToken({req})
const userObject = await getFullUserObject({token, sessionData})
```

In case if the token is expired, `getFullUserObject` is failing. When designing this solution I expected that my `jwt` and `session` callbacks will be triggered by `getSession` but in fact, they are not. 

Is there a fix to this flow and maybe I'm just doing something wrong?

Thank you

**Comment by Mosquid** [comment]
It seems like I should be proxying external API calls. It's something I initially tried to avoid but it looks like a common practice.

**Comment by balazsorban44** [comment]
session is a subset of token, so `getToken` should be enough here.

**Reply by Mosquid** [reply]
Yes, makes sense. But is there a way that will allow triggering `jwt` of `session` callback from `getServerSideProps`?

**Reply by balazsorban44** [reply]
getToken itself is a very simple method, that just decodes what's inside the cookie, and doesn't go through the lifecycles of getSession.

**Comment by balazsorban44** [comment]
When used server-side, getSession also expects the req object. https://next-auth.js.org/getting-started/client#server-side-example

and it should be calling both the jwt and session callbacks.
